Firestore/core/test/firebase/firestore/model/field_value_test.cc

```diff
@@ -73,6 +73,25 @@ MATCHER(IsNan, "a NaN") {
 
 }  // namespace
 
+TEST(FieldValueTest, ValueHelpers) {
+  // Validates that the Value helpers in testutil produce the right types
+  FieldValue bool_value = Value(true);
+  ASSERT_EQ(bool_value.type(), Type::Boolean);
+  EXPECT_EQ(bool_value.boolean_value(), true);
+
+  FieldValue int_value = Value(5);
+  ASSERT_EQ(int_value.type(), Type::Integer);
+  EXPECT_EQ(int_value.integer_value(), 5);
+
+  FieldValue long_value = Value(LONG_MAX);
```

Comment:
Would it be worth adding a test case that explicitly tests a value that's greater than 2^32? LONG_MAX fits the bill on linux/gcc, but presumably not under mac/clang (or else we would've run into this in the first place.)

Reply:
I don't think a larger value would have helped because on clang/mac that would have matched the `int64_t` overload just fine. The issue was the existence of the overload for `long` and `int64_t` when both were the same type. An additional literal doesn't help with that.

However, it also doesn't hurt, so I've added one.